ksp-processors/src/main/kotlin/graal/KordGraalProcessor.kt

```diff
@@ -0,0 +1,94 @@
+package dev.kord.ksp.graal
+
+import com.google.devtools.ksp.processing.*
+import com.google.devtools.ksp.symbol.KSAnnotated
+import com.google.devtools.ksp.symbol.KSClassDeclaration
+import dev.kord.ksp.GraalVisible
+import dev.kord.ksp.getSymbolsWithAnnotation
+import dev.kord.ksp.jvmBinaryName
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+
+class KordGraalProcessorProvider : SymbolProcessorProvider {
+    override fun create(environment: SymbolProcessorEnvironment): SymbolProcessor {
+        return KordGraalProcessor(
+            environment.codeGenerator, environment.logger
+        )
+    }
+}
+
+private val entries = mutableListOf<ReflectConfigEntry>()
+
+private class KordGraalProcessor(private val codeGenerator: CodeGenerator, private val logger: KSPLogger) :
+    SymbolProcessor {
+    override fun finish() {
+        flushEntries()
+        logger.info("KordGraalProcessor received finish signal")
+    }
+
+    override fun onError() {
+        logger.info("KordGraalProcessor received error signal")
+    }
+
+    override fun process(resolver: Resolver): List<KSAnnotated> {
+        logger.info("KordGraalProcessor got called, resolving annotations...")
+
+        entries += resolver
+            .getSymbolsWithAnnotation<Serializable>()
+            .filterIsInstance<KSClassDeclaration>()
+            .flatMap(::processClass)
+            .toList()
+
+        entries += resolver
+            .getSymbolsWithAnnotation<GraalVisible>()
+            .filterIsInstance<KSClassDeclaration>()
+            .map { ReflectConfigEntry(name = it.jvmBinaryName) }
+
+        logger.info("KordEnumProcessor finished processing annotations")
+
+        return emptyList() // we never have to defer any symbols
+    }
+
+    private fun flushEntries() {
+        if (entries.isNotEmpty()) {
+            val config = ReflectConfig(entries.distinctBy(ReflectConfigEntry::name))
+            val file = codeGenerator
+                .createNewFile(Dependencies.ALL_FILES, "META-INF.native-image", "reflect-config", "json")
```

Comment:
[the latest docs](https://www.graalvm.org/latest/reference-manual/native-image/metadata/) say that this file should be in `META-INF/native-image/<group.id>/<artifact.id>`, though i didn't try if this works